crates/ev-cli/src/commands/interact.rs

```diff
@@ -70,32 +60,142 @@ pub mod validators {
+        );
+        let length = name.len();
+        let regex_result = NAME_REGEX.is_match(name);
+        if !((length >= 2) & (length <= 40) & regex_result) {
+            return Err(ValidationError::InvalidFunctionName);
+        }
+        Ok(())
+    }
+
+    pub fn validate_function_language(language: &str) -> Result<(), ValidationError> {
+        lazy_static::lazy_static!(
+            static ref LANGUAGE_REGEX: Regex = Regex::new(r"^node@18|node@20|python@3\.9|python@3\.10|python@3\.11$").unwrap();
+        );
+        let regex_result = LANGUAGE_REGEX.is_match(language);
+        if !regex_result {
+            return Err(ValidationError::InvalidFunctionLanguage);
+        }
+
+        Ok(())
+    }
 }
 
-pub fn input(prompt: Option<String>, allow_empty: bool) -> Option<String> {
+pub fn input<T>(prompt: T, allow_empty: bool) -> String
+where
+    T: std::fmt::Display,
+{
     let theme = CliTheme::default();
     let mut input: Input<String> = Input::with_theme(&theme);
-    if let Some(prompt) = prompt {
-        input.with_prompt(prompt);
-    }
 
-    input.allow_empty(allow_empty).interact().ok()
+    match input
+        .with_prompt(prompt.to_string())
+        .allow_empty(allow_empty)
+        .interact()
+    {
+        Ok(input) => input,
+        Err(e) => {
+            eprintln!("Error reading user input : {}", e);
+            std::process::exit(1);
+        }
+    }
 }
 
-pub fn validated_input(
-    prompt: Option<String>,
+pub fn validated_input<T>(
+    prompt: T,
     allow_empty: bool,
     validator: Box<validators::GenericValidator>,
-) -> Option<String> {
+) -> Option<String>
+where
+    T: std::fmt::Display,
+{
     let theme = CliTheme::default();
     let mut input: Input<String> = Input::with_theme(&theme);
-    if let Some(prompt) = prompt {
-        input.with_prompt(prompt);
-    }
 
     input
+        .with_prompt(prompt.to_string())
         .allow_empty(allow_empty)
         .validate_with(validator)
         .interact()
         .ok()
 }
+
+pub fn select(options: &Vec<String>, default: usize, prompt: Option<String>) -> Option<usize> {
+    let theme = CliTheme::default();
+    let mut select_obj = Select::with_theme(&theme);
+    if let Some(prompt) = prompt {
+        select_obj.with_prompt(prompt);
+    }
+    select_obj.items(options).default(default).interact().ok()
+}
+
+/// To make quiet mode integration more simple
+/// OptionalProgressBar can be used - all functions called
+/// as normal, but will result in No-Ops during quiet mode
+///
+/// There may be an argument for implementing this using Deref
+/// To coerce OptPB to a PB silently, but unsure
+pub struct OptionalProgressBar {
+    bar: Option<ProgressBar>,
+}
+
+impl OptionalProgressBar {
+    pub fn new_spinner(quiet: bool) -> Self {
+        Self {
+            bar: if quiet {
+                None
+            } else {
+                Some(ProgressBar::new_spinner())
+            },
```

Comment:
This if else is awful to look at
```suggestion
            bar: quiet.then(ProgressBar::new_spinner),
```

Reply:
yea just lifted, noting all these comments for clean up pr